src/main/java/com/thealgorithms/maths/FibonacciNumber.java

```diff
@@ -0,0 +1,42 @@
+// Created by Pronay Debnath
+// Date:- 1/10/2003
+// Updated FibonnaciNumber.java using golden ratio formula
+// Explanation:- https://www.quickanddirtytips.com/articles/what-is-the-golden-ratio-and-how-is-it-related-to-the-fibonacci-sequence/
+
+import java.util.*;
+
+public class FibonacciNumber<T extends Number> {
+
+    // Function to find the nth Fibonacci number using the golden ratio formula
+    public static <T extends Number> int nthFibonacci(T n) {
+        // Calculate the square root of 5
+        double sqrt5 = Math.sqrt(5);
+
+        // Calculate the golden ratio (phi) and its conjugate (psi)
+        double phi = (1 + sqrt5) / 2;
+        double psi = (1 - sqrt5) / 2;
+
+        // Convert the input value to a double
+        double result = n.doubleValue();
+
+        // Calculate the nth Fibonacci number using the golden ratio formula
+        double fibonacci = (1 / sqrt5) * (Math.pow(phi, result) - Math.pow(psi, result));
```

Comment:
How _exact_ is this method? Can you prove that it gives the correct result for all values of `n`?

Reply:
I don't think it is very much exact as of now, I mean it can show the errors when given wrong input type, I have to look after the code and edit that once again

Reply:
I meant the following: if you compute the `n`-th Fibonacci number using this method, can you prove that the result is exactly the `n`-th Fibonacci number? Fibonacci sequence grows quite rapidly, there are some rounding errors etc.

Reply:
Maybe, it will show some errors as you have said, I have to check this again